pkg/runtime/wfengine/component.go

```diff
@@ -170,6 +177,10 @@ func (c *workflowEngineComponent) Get(ctx context.Context, req *workflows.GetReq
 	}
 
 	if metadata, err := c.client.FetchOrchestrationMetadata(ctx, api.InstanceID(req.InstanceID)); err != nil {
+		if err == ErrInstanceNotFound {
+			c.logger.Infof("Unable to get data on the instance, no such instance exists")
+			return nil, nil
```

Comment:
What will be the HTTP status code that's returned in this case? I don't think it's appropriate for an HTTP API to return 200 in this case. Is there a way for us to return an HTTP 404 without returning an error?